src/util/git_repos.rs

```diff
@@ -104,40 +104,23 @@ impl GitRepoManager {
     }
 
     /// Runs git fetch on the cached repo for the software specified by `software_id`, determines
-    /// whether `commit_or_tag` is a commit or tag, and if it is a commit, returns the commit and
-    /// the tags if the commit has any, and if it is a tag, returns the commit for that tag and any
-    /// tags for that commit.  If it's neither, returns Error::NotFound.  Returns an error for any
-    /// other failures
-    pub fn get_commit_and_tags_and_date_from_commit_or_tag(
+    /// the commit hash for `git_ref`, and gets any tags for that commit along with the commit date.
+    /// Returns the commit hash, tags, and commit date.  If `git_ref` not a valid git ref, returns
+    /// Error::NotFound.  Returns an error for any other failures
+    pub fn get_commit_and_tags_and_date_from_ref(
         &self,
         software_id: Uuid,
-        commit_or_tag: &str,
+        git_ref: &str,
     ) -> Result<(String, Vec<String>, NaiveDateTime), Error> {
```

Comment:
NaiveDateTime?

Reply:
It's a rust type for datetimes without timezones.  I think I originally picked it as the type I'd use for datetimes because of something I thought was a restriction in the DB library I'm using, but I think I was wrong about that.  I gotta look into it again because I think I can probably switch to regular DateTime.  I just need to actually set aside some time to do it.

Reply:
Interesting.  Time is always such a clusterfuck to represent. 

some *NaiveTime to do it.